### ISSUE TITLE
Incorrect behavior or documentation for comments in salt.states.pkgrepo.managed

### ISSUE THREAD

**Issue by craigafinch** [saltstack/salt#34963]
### Description of Issue/Question

`comments` in `pkgrepo.managed` does not behave as documented. The documentation states, "Anything supplied for this list will be saved in the repo configuration with a comment marker (#) in front." However, Salt does not behave this way; anything in the comments section is added WITHOUT a # in front. I am not sure if the behavior or the documentation is wrong, but they don't match.
### Steps to Reproduce Issue

This state:

```
epel:
  pkgrepo.managed:
    - humanname: Extra Packages for Enterprise Linux 7 - $basearch
    - mirrorlist: https://mirrors.fedoraproject.org/metalink?repo=epel-7&arch=$basearch
    - comments:
        - 'baseurl=http://download.fedoraproject.org/pub/epel/7/$basearch'
    - gpgcheck: 1
    - gpgkey: file:///etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7
    - priority: 3
```

Produces this repo file:

```
[epel]
priority=3
mirrorlist=https://mirrors.fedoraproject.org/metalink?repo=epel-7&arch=$basearch
name=Extra Packages for Enterprise Linux 7 - $basearch
gpgcheck=1
gpgkey=file:///etc/pki/rpm-gpg/RPM-GPG-KEY-EPEL-7
enabled=1
baseurl=http://download.fedoraproject.org/pub/epel/7/$basearch
```

Note that no # sign is placed in front of `baseurl`, so it is not actually a comment. You have to include the # sign in your state.
### Versions Report

```
Salt Version:
           Salt: 2016.3.1

Dependency Versions:
           cffi: Not Installed
       cherrypy: Not Installed
       dateutil: Not Installed
          gitdb: Not Installed
      gitpython: Not Installed
          ioflo: Not Installed
         Jinja2: 2.7.2
        libgit2: Not Installed
        libnacl: Not Installed
       M2Crypto: Not Installed
           Mako: Not Installed
   msgpack-pure: Not Installed
 msgpack-python: 0.4.6
   mysql-python: Not Installed
      pycparser: Not Installed
       pycrypto: 2.6.1
         pygit2: Not Installed
         Python: 2.7.5 (default, Nov 20 2015, 02:00:19)
   python-gnupg: Not Installed
         PyYAML: 3.11
          PyZMQ: 14.7.0
           RAET: Not Installed
          smmap: Not Installed
        timelib: Not Installed
        Tornado: 4.2.1
            ZMQ: 4.0.5

System Versions:
           dist: centos 7.2.1511 Core
        machine: x86_64
        release: 3.10.0-327.22.2.el7.x86_64
         system: Linux
        version: CentOS Linux 7.2.1511 Core
```

**Comment by Ch3LL** [comment]
@craigafinch I am able to replicate this behavior on 2016.3.1 and 2015.8.10. Looks like we neeed to ensure comments are added apropriately with `#`.

**Comment by derekmaciel** [comment]
Does this bug exist on non-yum systems? It looks like currently `states.pkgrepo` does nothing with `comments` and merely defers it to the system's respective `pkg` module.

**Comment by craigafinch** [comment]
I only have Yum systems to test on, so I am unable to help with that question.

**Comment by oliver-dungey** [comment]
I've just bumped into this one on 2016.3.3

**Comment by connordelacruz** [comment]
I've encountered this issue as well on 2016.11.5. As a workaround, I got it to work by surrounding each item in the `- comments` list with quotes and manually added the # at the beginning. e.g.:
```
- comments:
  - '# line 1'
  - '# line 2'
```

**Comment by nhavens** [comment]
This issue should be closed, as it was resolved by the PR #41820 mentioned above. However, there is a regression in v2017.7.2 that has caused the issue to reoccur. I have opened Issue #44299 to discuss the new problem.

**Comment by gtmanfred** [comment]
Thanks
